libdirections-session/src/main/java/com/mapbox/navigation/directions/session/MapboxDirectionsSession.kt

```diff
@@ -54,12 +63,18 @@ class MapboxDirectionsSession(private val router: Router) : DirectionsSession {
     }
 
     private fun requestRoute() {
-        if (::origin.isInitialized && ::waypoints.isInitialized) {
-            router.cancel()
-            currentRoute = null
-            router.getRoute(origin, waypoints) {
-                currentRoute = it
+        router.cancel()
+        currentRoute = null
```

Comment:
We'll either need to call `onRouteChanged` here or do not nullify. Possibly not nullifying and just keeping the old route would be better if the request fails?

Reply:
Don't think have invalid route better, than don't have any route

Reply:
I don't even think that `MapboxDirectionsSession` must keep any state like `Route` or anything else

Reply:
We'd need to additionally call `routeObservers.forEach { it.onRouteChanged(null) }` then, right after nullifying the field. We should add an appropriate test as well.

Reply:
I believe it'll confuse customer. IMO in `onFailure` callback set `currentRoute = null`